src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -380,141 +384,101 @@ protected function assignJoins(QueryBuilder $queryBuilder, array $joins = null)
     }
 
     /**
-     * Adds where statements for in-clauses.
-     *
-     * @param array $inFields
-     * @param array $inValues
+     * {@inheritdoc}
      */
-    protected function addIns(array $inFields, array $inValues)
+    public function createWhereExpression(AbstractFieldDescriptor $fieldDescriptor, $value, $comparator)
     {
-        $inParts = [];
-        foreach ($inFields as $inField) {
-            $inPart = $inField->getSelect() . ' IN (:' . $inField->getName() . ')';
-            $this->queryBuilder->setParameter($inField->getName(), $inValues[$inField->getName()]);
-
-            // null values
-            if (array_search(null, $inValues[$inField->getName()])) {
-                $inPart .= ' OR ' . $inField->getSelect() . ' IS NULL';
-            }
-
-            $inParts[] = $inPart;
+        if ($fieldDescriptor instanceof AbstractDoctrineFieldDescriptor) {
```

Comment:
Usually we check for the error first, so I would inverted this statement, and put the throw statement into the if.

```php
if (!$fieldDescriptor instanceof AbstractDoctrineFieldDescriptor) {
    throw new InvalidArgumentException(...);
}

return new DoctrineWhereExpression(...);
```